### PR TITLE
Configure DWO to not process proxy settings for workspaces Che manages

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
If Che detects proxy settings on a cluster (either through configuration or the OpenShift cluster-wide proxy), configure the DevWorkspace Operator to not use any proxy settings internally for workspaces. 

This is necessary to allow the `proxy-settings` configmap provisioned by Che to actually be used in workspaces; otherwise, DWO will set the proxy environment variables in the workspace deployment directly, overriding env vars set from the configmap.

In addition, this PR updates the proxy env configmap to include lowercase versions of environment variables (i.e. `http_proxy`, `https_proxy`, `no_proxy`) to try to ensure compatibility with all tools (e.g. `curl` _only_ uses `http_proxy`, not `HTTP_PROXY`)

~This PR is marked WIP as it depends on changes from https://github.com/devfile/devworkspace-operator/pull/1147; once DWO v0.23 is released, the first commit in this PR needs to be updated to use 0.23 instead of a main-branch commit.~

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
Closes https://github.com/eclipse/che/issues/22370

### How to test this PR?
Changes from this PR are pushed to `quay.io/amisevsk/che-operator:proxy-config`

To test, 
1. Start an OpenShift cluster with a cluster-wide proxy configured
2. Install Che + DWO: `chectl server:deploy -p openshift --olm-channel next`
3. Update the eclipse-che CSV (`eclipse-che.<ver>.next`) to use `quay.io/amisevsk/che-operator:proxy-config` for the Che Operator image
4. Start a workspace

Once workspace is running, verify
1. DevWorkspace operator config is configured to not set proxy variables:
    ```
    oc get dwoc -n eclipse-che devworkspace-config -o yaml
    (...)
      config:
        routing:
          proxyConfig:
            httpProxy: ""
            httpsProxy: ""
            noProxy: ""
    (...)
    ```
2. The workspace's deployment does not set proxy environment variables directly
3. Proxy variables are still present in workspace, mounted from configmap
    ```
    $ env | grep -i proxy
    HTTP_PROXY=http://<snip>@10.0.15.91:3128/
    https_proxy=http://<snip>@10.0.15.91:3128/
    http_proxy=http://<snip>@10.0.15.91:3128/
    no_proxy=.cluster.local,#etc
    NO_PROXY=.cluster.local,#etc
    HTTPS_PROXY=http://<snip>@10.0.15.91:3128/
    ```
4. Changes to proxy settings in CheCluster propagate down to workspaces and aren't ignored due to DWO's conflicting config

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Development resource are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-development-resources)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by amisevsk** [comment]
PR is now ready for review

File: controllers/usernamespace/controller_test.go [hunk 1]
```diff
@@ -341,7 +341,7 @@ func TestCreatesDataInNamespace(t *testing.T) {
 		assert.Equal(t, "true", proxySettings.GetLabels()[dwconstants.DevWorkspaceMountLabel],
 			"proxy settings should be labeled as mounted")
 
-		assert.Equal(t, 1, len(proxySettings.Data), "Expecting just 1 element in the default proxy settings")
+		assert.Equal(t, 2, len(proxySettings.Data), "Expecting just 1 element in the default proxy settings")
```
Comment (tolusha) [inline controllers/usernamespace/controller_test.go:344]:
```suggestion
		assert.Equal(t, 2, len(proxySettings.Data), "Expecting 2 elements in the default proxy settings")
```

File: pkg/deploy/dev-workspace-config/dev_workspace_config.go [hunk 2]
```diff
@@ -104,6 +105,13 @@ func updateWorkspaceConfig(cheCluster *chev2.CheCluster, operatorConfig *control
 
 	updateWorkspaceImagePullPolicy(devEnvironments.ImagePullPolicy, operatorConfig.Workspace)
 
+	if ctx.Proxy.HttpProxy != "" || ctx.Proxy.HttpsProxy != "" {
```
Comment (tolusha) [inline pkg/deploy/dev-workspace-config/dev_workspace_config.go:108]: 1. Could you add some comments explaining why we have to clean up proxy config, please?
2. There is something that I don't understand. If proxy is set, then we have to clean up to DWO proxy config. If proxy is not set, then we do nothing. What is the point of DWO proxy config then?

Comment (amisevsk) [reply pkg/deploy/dev-workspace-config/dev_workspace_config.go:108]: Yup, I agree some comments here would be helpful.

As for why DWO supports configuring the proxy, the answer is twofold: first, it exists as an option because we implemented it (way back in v0.12, before proxy was supported in Che Operator); second, it's a feature that's still used by other consumers of DevWorkspaces, such as Web Terminal.

Che introduced its own implementation of managing proxy settings for DevWorkspaces, but the two implementations collide -- Che supports the proxy by creating a `che-proxy-settings` configmap which is mounted as env vars, and so needs to disable built-in DWO proxy configuration/detection. 

Basically, what we're doing here is telling DWO "we'll handle setting up proxy environment variables, you shouldn't do it for this workspace".

**Review by tolusha — APPROVED** [review]
Thank you for explanation.
Please merge.
